pkg/external/external.go

```diff
@@ -237,3 +244,31 @@ func (alert MemProtAlert) String() string {
 		return "Unknown alert"
 	}
 }
+
+//for supporting networking events in tracee rules
+type PktMeta struct {
+	SrcIP    string `json:"srcIP"`
+	DestIP   string `json:"destIP"`
+	SrcPort  uint16 `json:"srcPort"`
+	DestPort uint16 `json:"destPort"`
+	Protocol uint8  `json:"protocol"`
+}
+
+type FunctionBasedPacket struct {
+	LocalIP     string  `json:"localIP"`
+	RemoteIP    string  `json:"remoteIP"`
+	LocalPort   uint16  `json:"localPort"`
+	RemotePort  uint16  `json:"remotePort"`
+	Protocol    uint8   `json:"protocol"`
+	_           [3]byte //padding
+	TcpOldState uint32  `json:"tcpOldState"`
+	TcpNewState uint32  `json:"tcpNewState"`
+	_           [4]byte //padding
+	SockPtr     uint64  `json:"sockPtr"`
+}
+
+type DnsQueryData struct {
```

Comment:
It can be but I think it will be better as struct to save more place for other arguments.

Note that we save the `dns_request` event arguments as `DnsQueryData` for each arg.
In that case we support until 7 questions in event, so if ill expend it from struct to field it will be up to 2 questions